apps/manager_app/auth_card.c

```diff
@@ -0,0 +1,542 @@
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR
+ * ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ *
+ * "Commons Clause" License Condition v1.0
+ *
+ * The Software is provided to you by the Licensor under the License,
+ * as defined below, subject to the following condition.
+ *
+ * Without limiting other conditions in the License, the grant of
+ * rights under the License will not include, and the License does not
+ * grant to you, the right to Sell the Software.
+ *
+ * For purposes of the foregoing, "Sell" means practicing any or all
+ * of the rights granted to you under the License to provide to third
+ * parties, for a fee or other consideration (including without
+ * limitation fees for hosting or consulting/ support services related
+ * to the Software), a product or service whose value derives, entirely
+ * or substantially, from the functionality of the Software. Any license
+ * notice or attribution required by the License must also include
+ * this Commons Clause License Condition notice.
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+#include "controller_tap_cards.h"
+#include "events.h"
+#include "lv_task.h"
+#include "manager_api.h"
+#include "manager_app.h"
+#include "manager_app_priv.h"
+#include "nfc.h"
+#include "status_api.h"
+#include "ui_confirmation.h"
+#include "ui_delay.h"
+#include "ui_instruction.h"
+
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+#define HANDLE_P0_EVENTS()
+#define CHALLENGE_SIZE 32
+#define UPDATE_AUTH_CARD_FLOW_STATUS(status)                                   \
+  auth_card_ctx.flow_status = status;                                          \
+  core_status_set_flow_status(status)
+
+#define SIGN_SERIAL_BEEP_COUNT(pair_card_required)                             \
+  (false == pair_card_required) ? 3 : 2
+#define SIGN_CHALLENGE_BEEP_COUNT(pair_card_required)                          \
+  (false == pair_card_required) ? 2 : 1
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+typedef struct {
```

Comment:
Is there any way in which we can make this a local variable and pass it through pointers in the different handlers?

It would help in reducing overall global variables used in the firmware